be/dtm0_log.h

```diff
@@ -0,0 +1,522 @@
+ * fields in this structure are:
+ * - dlr_dtx: This stores dtx information related to dtm0 client.
+ * - dlr_txd: This stores the states of the participants.
+ * - dlr_payload: This stores the original cas_op (cas request).
+ */
+
+struct m0_dtm0_log_rec {
+	struct m0_dtm0_dtx     dlr_dtx;
+	struct m0_dtm0_tx_desc dlr_txd;
+	uint64_t               dlr_magic;
+	union  {
+		struct m0_be_list_link dlr_link;    /**< link into m0_be_dtm0_log::tlist */
+		struct m0_tlink        dlr_tlink;
+	} u;
+	struct m0_buf          dlr_payload;
+};
+
+/**
+ * @b  m0_be_dtm0_log_rec structure
+ *
+ * A DTM0 log is represented by m0_be_dtm0_log structure. The important
+ * fields in this structure are:
+ * - dl_is_persistent: This is a flag to distinguish between a volatile
+ * (client-side) log and a persistent (server-side) log.
+ * - dl_cs: A pointer to the type of clock used to generate the timestamps
+ * for the log records.
+ */
+
+struct m0_be_dtm0_log {
+	/** Indicates if the structure is a persistent or volatile */
+	bool                     dl_is_persistent;
+	/** dl_lock protects access to the dl_list/dl_tlist. This lock
+	 *  should be held before performing any operations on the log
+	 *  involving a log record search or insert. */
+	struct m0_mutex          dl_lock;
+	struct m0_be_seg	*dl_seg;
+	/** DTM0 clock source */
+	struct m0_dtm0_clk_src	*dl_cs;
+	union {
+		/** Persistent list, used if dl_is_persistent */
+		struct m0_be_list	*dl_list;
+		/** Volatile list, used if !dl_is_persistent */
+		struct m0_tl		*dl_tlist;
+	} u;
+};
+
+/**
+ * @b Typical call flow for calling DTM0 log interface routines:
+ *
+ * ** Initialisation/Finalisation for volatile fields of the log
+ * ** Create a persistent log or just allocate a volatile (in-memory) log
+ *   m0_be_dtm0_log_create / m0_be_dtm0_log_alloc
+ * - m0_be_dtm0_log_init()
+ * ** Preparation phase
+ * - m0_be_dtm0_log_credit()
+ * ** Normal operation phase
+ * - m0_be_dtm0_log_update()
+ * - m0_be_dtm0_log_prune()
+ * - m0_be_dtm0_log_find()
+ */
+
+/**
+ * Allocate an m0_be_dtm0_log. This call will allocate a dtm0 volatile log.
+ * We need to call this routine once before we can perform any operations on .
+ *
+ * @pre out != NULL.
+ * @post *log is allocated and ready to be initialized.
+ *
+ * @param log Pointer to variable in which to return the allocate memory
+ *        address.
+ *
+ * @return 0 on success. Anything else is a failure.
+ */
+M0_INTERNAL int m0_be_dtm0_log_alloc(struct m0_be_dtm0_log **log);
+
+/**
+ * Initialize a dtm0 log. We need to call this routine once before we can
+ * perform any operations on it.
+ *
+ * @pre log != NULL;
+ * @pre cs != NULL;
+ * @post *log is initialized and ready to be used
+ *
+ * @param log Pointer to a m0_be_dtm0_log that has to be initialized.
+ * @param cs Pointer to a clock source. This will be used for comparisons
+ *        involving log record timestamps.
+ * @param isvstore A flag to indicate whether the dtm0 log that we are trying
```

Comment:
isvstore?
is_plog?

We need to sync the comments and function definition.

Reply:
Sorry, missed it. Done. Thanks.